src/Extensions.Hosting.AsyncInitialization/Hosting/AsyncInitializationHostExtensions.cs

```diff
@@ -32,5 +32,35 @@ public static async Task InitAsync(this IHost host, CancellationToken cancellati
 
             await rootInitializer.InitializeAsync(cancellationToken);
         }
+
+
+        /// <summary>
+        /// Initializes and runs the application, by first calling all registered async initializers.
+        /// Upon completion, any registered initializers implementing <see cref="IAsyncTeardown"/> are called.
+        /// </summary>
+        /// <param name="host">The host.</param>
+        /// <param name="cancellationToken">Optionally propagates notifications that the operation should be cancelled</param>
+        /// <returns>A task that represents the initialization completion.</returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="InvalidOperationException"></exception>
+        /// <exception cref="OperationCanceledException"></exception>
+        public static async Task RunInitAsync(this IHost host, CancellationToken cancellationToken = default)
```

Comment:
I feel like the name of this method could be more explicit.
`RunWithInitializersAndTeardownAsync` is a mouthful, though...

Maybe `InitAndRunAsync`? It doesn't mention teardown, but maybe it doesn't need to, since the documentation clearly mentions it.

Reply:
Yes, I was struggling with the naming as well. An alternative could be `RunWithInitAsync`.
It leans towards Host.RunAsync() and suggests that the call is wrapped with initialization.
Hope you agree :-)
Done.

Reply:
I could live with `RunWithInitAsync`, but I like `InitAndRunAsync` better

Reply:
Okay, changed it to ItintAndRunAsync().